pkg/deploy/controller/deploymentconfig/metrics.go

```diff
@@ -0,0 +1,63 @@
+package deploymentconfig
+
+import (
+	"strings"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// DeploymentConfigControllerSubsystem is how this controller is represented in
+// prometheus metrics.
+const DeploymentConfigControllerSubsystem = "deploymentconfig_controller"
+
+var (
+	deploymentConditionsCounter = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Subsystem: DeploymentConfigControllerSubsystem,
+			Name:      "current_conditions",
+			Help:      "Gauge of current number of deployment conditions per condition type, reason and status",
+		},
+		[]string{"type", "reason", "status"},
+	)
+)
+
+func init() {
+	prometheus.MustRegister(deploymentConditionsCounter)
+}
+
+// updateConditionsMetrics updates prometheus gauge counters based on the delta between
+// old and new deployment conditions.
+func updateConditionsMetrics(old, new []deployapi.DeploymentCondition) {
+	conditionToLabelValues := func(c deployapi.DeploymentCondition) []string {
+		return []string{
+			strings.ToLower(string(c.Type)),
+			strings.ToLower(string(c.Reason)),
+			strings.ToLower(string(c.Status)),
+		}
+	}
+	for _, n := range new {
+		found := false
+		for _, o := range old {
+			if n == o {
```

Comment:
This will fail in the case of a Progressing condition where the only difference between the new and the old condition is the lastTransitionTime, right?